engine/gfx/color.asm

```diff
@@ -935,7 +935,14 @@ LoadMapPals:
 	ldh [rSVBK], a
 
 .got_pals
+	ld hl, wPalFlags
+	bit MAP_CONNECTION_PAL_F, [hl]
+	jr nz, .skip_clearing_obj_pals
 	farcall ClearSavedObjPals
+.skip_clearing_obj_pals
+
+	ld hl, wPalFlags
+	res MAP_CONNECTION_PAL_F, [hl]
```

Comment:
Would this work instead? Note that `res` does not disturb the flags set by `bit`.
```suggestion
	ld hl, wPalFlags
	bit MAP_CONNECTION_PAL_F, [hl]
	res MAP_CONNECTION_PAL_F, [hl]
	jr nz, .skip_clearing_obj_pals
	farcall ClearSavedObjPals
.skip_clearing_obj_pals
```